tests/jsTests/indexSpec.js

```diff
@@ -0,0 +1,283 @@
+	});
+});
+
+/**
+ * Suite : Tests for success response modal
+ * Description : This is a test suite for checking when the response is a success
+ */
+
+describe("Tests for success response modal", function() {
+	var msg = 'This is a test message when the response is a success';
+	beforeEach(function() {
+		jasmine.clock().install();
+		setFixtures('<div class="element"></div>\
+		<div class="processing"></div>\
+		<div class="ui modal">\
+			<div class="header"></div>\
+			<div class="content"></div>\
+			<div class="actions">\
+				<div class="ui cancel green button">Close</div>\
+			</div>\
+		</div>');
+		$('.ui.modal').hide();
+
+		// mock modal show call
+		spyOn($.fn, "modal").and.returnValue($('.ui.modal').show());
+		showResponse($('element'), { response: true, message: msg });
+		jasmine.clock().tick(4000);
+	});
+
+	afterEach(function() {
+		jasmine.clock().uninstall();
+	});
+
+	it("should be visible", function() {
+		expect($('.content')).toContainText(msg);
+		expect($('.ui.modal')).toBeVisible();
+	});
+});
+
+/**
+ * Suite : Tests for failed response modal
+ * Description : This is a test suite for a failed response modal
+ */
+describe("Tests for failed response modal", function() {
+	var msg = 'This is a test message for a failed response modal';
+	beforeEach(function() {
+		jasmine.clock().install();
+		setFixtures('<div class="element"></div>\
+		<div class="ui modal">\
+			<div class="header"></div>\
+			<div class="content"></div>\
+			<div class="actions">\
+				<div class="ui cancel green button">Close</div>\
+			</div>\
+		</div>');
+		$('.ui.modal').hide();
+
+		// mock modal show call
+		spyOn($.fn, "modal").and.returnValue($('.ui.modal').show());
+		showResponse($('element'), { response: false, message: msg });
+		jasmine.clock().tick(4000);
+	});
+
+	afterEach(function() {
+		jasmine.clock().uninstall();
+	});
+
+	it("should be visible", function() {
+		expect($('.content')).toContainText(msg);
+		expect($('.header')).toContainText('Are you lost?');
+		expect($('.ui.modal')).toBeVisible();
+	});
+});
+
+/**
+ * Suite : Tests for password score
+ * Description : This is a test suite for checking password score
+ */
+describe("Tests for password score", function() {
+	it("should be visible", function() {
+		var passwords = {
+			0: {
+				'text': 'psswd',
+				'score': 0
+			},
+			1: {
+				'text': 'password',
+				'score': 2
+			},
+			2: {
+				'text': 'Password',
+				'score': 3
+			},
+			3: {
+				'text': 'password1',
+				'score': 3
+			},
+			4: {
+				'text': 'password#',
+				'score': 3
+			},
+			5: {
+				'text': 'Password1',
+				'score': 4
+			},
+			6: {
+				'text': 'Password#',
+				'score': 4
+			},
+			7: {
+				'text': 'Password#1',
+				'score': 5
+			},
+			8: {
+				'text': 'Password3#',
+				'score': 5
+			}
+		}
+		for (var k in passwords) {
+			if (passwords.hasOwnProperty(k)) {
+				var s = passwordScore(passwords[k]['text']);
+				expect(s['score']).toBe(passwords[k]['score']);
+			}
```

Comment:
@MehaKaushik I will fix this. Error messages can change in future, so I will test the error counts.